lms/static/scripts/frontend_apps/utils/dashboard/hooks.ts

```diff
@@ -47,9 +47,16 @@ export function useDashboardFilters(): UseDashboardFilters {
         newQueryParams.student_id = studentIds;
       }
 
-      navigate(recordToQueryString(newQueryParams), { replace: true });
+      // The base URL is represented by '/', but that adds a trailing slash to
+      // the URL which is not recognized by the server router. To work around
+      // that, we replace it with an "empty" path.
+      const normalizedLocation = location === '/' ? '' : location;
```

Comment:
Can you provide an example situation where this causes a problem? Also, is there a bug in `navigate` here that we need to report or is wouter's behavior correct?

Reply:
I think this is somewhat expected, or at least a common behavior (although not super intuitive), as I have the feeling I have faced this with other client-side routing libraries.

For some reason, these libraries treat `"/"` and `""` interchangeably. So if your route structure looks like this:

```
<Route path="/dashboard" nest>
  <Route path="/foo">
    <InFoo />
  </Route>
  <Route path="">
    <InHome />
  </Route>
</Route>
```

Visiting `/dashboard` or `/dashboard/` will render `<InHome />` in both cases. And calling `useLocation()` inside `InHome` component, will return `/` in both cases.

However, `/dashboard/foo` will match the first route and render `<InFoo />`, but `/dashboard/foo/` will not match anything.

This shouldn't be a problem initially, as the client-side routing works, but if you reload the page when in `/dashboard/`, the server will return a 404, because the route does not match there.

That said, perhaps we should bring this up in wouter repository, to know if this is expected and understand why it works like this, or to bring it to their attention if it's in fact a bug.

Reply:
Just opened a discussion there https://github.com/molefrog/wouter/discussions/472

Reply:
> Also, is there a bug in navigate here that we need to report or is wouter's behavior correct?

They posted a reply there. It seems this is the expected behavior.